library/std/src/pipe.rs

```diff
@@ -1,20 +1,66 @@
-//! Module for anonymous pipe
+//!  A cross-platform anonymous pipe.
 //!
-//! ```
-//! #![feature(anonymous_pipe)]
+//! This module provides support for anonymous OS pipes, like [pipe] on Linux or [CreatePipe] on
+//! Windows, which can be used as synchronous communication channels between related processes.
+//!
+//! # Behavior
+//!
+//! A pipe can be thought of as a bounded, interprocess [`mpsc`](crate::sync::mpsc), provided by
+//! the OS, with a platform-dependent capacity. In particular:
```

Comment:
It would be good to say what a pipe is first and then point to MPSC. Maybe something like:

> A pipe is a unidirectional data channel that can be used to communicate among two or more processes, provided by the OS. It can be thought of as an interprocess [`mpsc`](crate::sync::mpsc).
>
> The channel is bounded with limited platform-dependent capacity. The following rules are observed: